cairo/tests/utils/strategies.py

```diff
@@ -203,29 +212,56 @@ def tuple_strategy(thing):
 ).map(lambda x: Message(**x))
 
 
-evm_lite = st.fixed_dictionaries(
-    {
-        "pc": uint,
-        "stack": stack_strategy(Stack[U256]),
-        "memory": memory_lite,
-        "code": st.just(b""),
-        "gas_left": st.integers(min_value=0, max_value=BLOCK_GAS_LIMIT).map(Uint),
-        "env": st.from_type(Environment),
-        "valid_jump_destinations": st.just(set()),
-        "logs": st.just(()),
-        "refund_counter": st.just(0),
-        "running": st.booleans(),
-        "message": message_lite,
-        "output": st.just(b""),
-        "accounts_to_delete": st.just(set()),
-        "touched_accounts": st.just(set()),
-        "return_data": st.just(b""),
-        "error": st.none() | st.from_type(EthereumException),
-        "accessed_addresses": st.just(set()),
-        "accessed_storage_keys": st.just(set()),
-    }
-).map(lambda x: Evm(**x))
+BLOCK_HASHES_LIST = [Hash32(Bytes32(bytes([i] * 32))) for i in range(256)]
+
+environment_lite_block = st.integers(
+    min_value=0
+).flatmap(  # Generate block number first
+    lambda number: st.builds(
+        Environment,
+        caller=address,
+        block_hashes=st.lists(
+            st.sampled_from(BLOCK_HASHES_LIST),
+            min_size=min(number, 256),  # number or 256 if number is greater
+            max_size=min(number, 256),
+        ),
+        origin=address,
+        coinbase=address,
+        number=st.just(Uint(number)),  # Use the same number
+        base_fee_per_gas=uint,
+        gas_limit=uint,
+        gas_price=uint,
+        time=uint256,
+        prev_randao=bytes32,
+        state=st.from_type(State),
+        chain_id=uint64,
+        excess_blob_gas=uint64,
+        blob_versioned_hashes=st.from_type(Tuple[VersionedHash, ...]),
```

Comment:
use st.tuples(st.from_type(VersionedHash), max_size=MAX_STORAGE_KEY_SET_SIZE) once https://github.com/kkrt-labs/keth/pull/375/files is merged